src/game/TacticalAI/DecideAction.cc

```diff
@@ -1749,7 +1749,7 @@ INT8 DecideActionRed(SOLDIERTYPE *pSoldier, UINT8 ubUnconsciousOK)
 
 			return(AI_ACTION_TOSS_PROJECTILE);
 		}
-		else		// toss/throw/launch not possible
+		else if (BestThrow.bWeaponIn != NO_SLOT)		// toss/throw/launch not possible
```

Comment:
1. What about the case of !BestThrow.ubPossible && BestThrow.bWeaponIn == NO_SLOT? It's not handled now, but went into this block before.
2. The comment is now confusing, since taken alone it looks wrong.

Reply:
It will continue to the code after the block, like it should (no item = no range).
I'll move the check to the inner `if`, that should take care of the confusing comment(s).

Reply:
Yeah, much better, thanks.